src/libraries/Core/Microsoft.Agents.Authentication/ConnectionSettingsBase.cs

```diff
@@ -12,24 +12,26 @@ namespace Microsoft.Agents.Authentication
     /// </summary>
     public abstract class ConnectionSettingsBase : IConnectionSettings
     {
-        protected ConnectionSettingsBase(IConfigurationSection configurationSection)
+        protected ConnectionSettingsBase() { }
+
+        protected ConnectionSettingsBase(IConfigurationSection msalConfigurationSection)
         {
-            if (configurationSection != null && configurationSection.Exists())
+            if (msalConfigurationSection != null && msalConfigurationSection.Exists())
             {
-                ClientId = configurationSection.GetValue<string>("ClientId", null);
-                Authority = configurationSection.GetValue<string>("AuthorityEndpoint", null);
-                TenantId = configurationSection.GetValue<string>("TenantId", null);
-                Scopes = configurationSection.GetSection("Scopes")?.Get<List<string>>();
+                ClientId = msalConfigurationSection.GetValue<string>("ClientId", null);
+                Authority = msalConfigurationSection.GetValue<string>("AuthorityEndpoint", null);
+                TenantId = msalConfigurationSection.GetValue<string>("TenantId", null);
+                Scopes = msalConfigurationSection.GetSection("Scopes")?.Get<List<string>>();
             }
             else
             {
-                if (configurationSection != null)
+                if (msalConfigurationSection != null)
                 {
-                    throw new ArgumentException($"Authentication configuration section {configurationSection.Key}, not Found.");
+                    throw new ArgumentException($"Authentication configuration section {msalConfigurationSection.Key}, not Found.");
```

Comment:
Add Error Wrapper here? ( there is a common function creating exceptions of a type)

Reply:
Yes.  You originally added Errors there, and used a different way, and not throught.  I'll take care of it